src/Integration.UnitTests/Telemetry/TelemetryManagerTests.cs

```diff
@@ -452,5 +452,160 @@ private static void CheckExpectedLanguages(TelemetryData data, params string[] e
+        [DataRow("2020-06-01T04:30:00.000+00:00", "2020-06-01T06:30:00.000+00:00", 0, false)] // same day -> don't update
+        [DataRow("2020-05-31T20:30:00.000+00:00", "2020-06-01T06:30:00.000+00:00", 0, true)] // different day -> update
+        [DataRow("2020-06-01T04:30:00.000+00:00", "2020-08-23T02:30:00.000+08:00", 8, true)] // many days later -> update
+        public void Update_IfNotSameDayAsLastUpdate_DaysOfUseIsIncremented(string lastSavedDate, string now, int utcOffset, bool expectedIncrement)
+        {
+            var lastSavedTimeUTC = DateTimeOffset.Parse(lastSavedDate);
+            var nowDateTimeOffset = DateTimeOffset.Parse(now);
+            const int initialDaysOfUse = 2;
+
+            // Arrange
+            var telemetryData = new TelemetryData
+            {
+                InstallationDate = lastSavedTimeUTC.AddDays(-100), // anything, as long as it's earlier
+                LastSavedAnalysisDate = lastSavedTimeUTC,
+                NumberOfDaysOfUse = initialDaysOfUse
+            };
+            telemetryRepositoryMock.Setup(x => x.Data).Returns(telemetryData);
+
+            var mockTimeProvider = CreateMockTimeProvider(nowDateTimeOffset, utcOffset);
+
+            var testSubject = CreateManager(mockTimeProvider.Object);
+
+            // Act
+            testSubject.Update();
+
+            // Assert
+            mockTimeProvider.Verify(x => x.Now, Times.Once);
+            mockTimeProvider.Verify(x => x.LocalTimeZone, Times.Once);
+
+            if (expectedIncrement)
+            {
+                telemetryData.LastSavedAnalysisDate.Should().Be(nowDateTimeOffset);
+                telemetryData.NumberOfDaysOfUse.Should().Be(initialDaysOfUse + 1);
+                this.telemetryRepositoryMock.Verify(x => x.Save(), Times.Once);
+            }
+            else
+            {
+                telemetryData.LastSavedAnalysisDate.Should().Be(lastSavedTimeUTC);
+                telemetryData.NumberOfDaysOfUse.Should().Be(initialDaysOfUse);
+                this.telemetryRepositoryMock.Verify(x => x.Save(), Times.Never);
+            }
+        }
+
+
+        [TestMethod]
+        [DataRow(0, 0, 1)] // first check on day of initial installation -> increment days of use
+        [DataRow(0, 1, 1)] // second and subsequent check on day of installation -> can't be incremented above 1
+
+        [DataRow(4, 4, 5)] // used every day since installation, already checked and incremented on day 4 -> incremented
+        [DataRow(4, 5, 5)] // used every day since installation, second and subsequent checks on day 4 -> not capped or incremented
+
+        [DataRow(1001, 100, 101)] // not used every day -> incremented and saved
+
+        [DataRow(1, 100, 2)] // bad data - should be capped at (days since installation + 1)
+        [DataRow(999, 1001, 1000)] // bad data - should be capped at (days since installation + 1)
+        public void Update_DaysOfUseIsCorrectlyCalculated(int daysSinceInstallation, int daysOfUse,
```

Comment:
This is the new test for the bug fix. The `days_of_use` can be incremented at to at most `days_since_installation`.
If it's greater than that value then it's capped.